### PR TITLE
Add Validations to Setters

### PR DESCRIPTION
**Pull request by Casanova569**
**Summary**
The card it is considered approved if the pre-conditions are placed and the app compile.

**Trello Card**
[Agregar validaciones de precondiciones a los setters](https://trello.com/c/CAyeDWLO/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters)

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -89,62 +90,72 @@ public String getAuthor() {
...
+		Preconditions.checkNotNull(year, "Year cannot be null");
 		this.year = year;
 	}
 
 	public Integer getPages() {
 		return this.pages;
 	}
 	public void setPages(Integer pages) {
+		Preconditions.checkNotNull(pages, "Pages cannot be null");
+		Preconditions.checkArgument(pages < 0, "Quantity of pages cannot be negative or zero");
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java]: Careful here, it should be `pages >= 0` 😃

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -89,62 +90,72 @@ public String getAuthor() {
...
 		this.pages = pages;
 	}
 
 	public String getIsbn() {
 		return this.isbn;
 	}
 	public void setIsbn(String isbn) {
+		Preconditions.checkNotNull(isbn);
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java]: Maybe you could add your own custom message :)

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -89,62 +90,72 @@ public String getAuthor() {
...
 		this.isbn = isbn;
 	}
 
 	public List<Users> getUsers() {
 		return (List<Users>) Collections.unmodifiableCollection(this.users);
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java]: I think you will need to switch `unmodifiableCollection` for `unmodifiableList`

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
Just a few tips! 😃 Also, remember that it's a good practice to isolate response messages as constants :)

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -91,62 +95,77 @@ public String getAuthor() {
...
 		this.isbn = isbn;
 	}
 
-	public List<User> getUsers() {
-		return (List<User>) Collections.unmodifiableCollection(this.users);
-	}
-	public void setUsers(User user) {
+	public void addUsers(User user) {
+		Preconditions.checkNotNull(user, "The user to add cannot be null");
+		if (this.users.contains(user))
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java:161]: You can also use Preconditions here 😄 just a note for the future :)

**Review by Jimenarosello — APPROVED** [review]
Nice job 😊

**Review by rodrigofranc — APPROVED** [review]

**Pull request merged**